### ISSUE TITLE
KVM VirtIO drive serial number not found

### ISSUE THREAD

**Issue by phillxnet** [rockstor/rockstor-core#673]
Originally though to be related to issue #666 but that one turned out to be a esxi client bug.

If one creates drives in kvm that use the VirtIO disk bus the serial number is not visible to Rockstor.

This I believe is due to Rockstor using the following command in btrfs.py to gain drive info:-
'''
/usr/bin/lsblk -P -o NAME,MODEL,SERIAL,SIZE,TRAN,VENDOR,HCTL,TYPE,FSTYPE,LABEL,UUID
'''
But this returns the following for the vda block device:-

NAME="vda" MODEL="" SERIAL="" SIZE="2G" TRAN="" VENDOR="0x1af4" HCTL="" TYPE="disk" FSTYPE="" LABEL="" UUID=""

So no serial number returned by lsblk but /etc/libvirt/qemu/rockstor3.7.xml contains the VMM configured serial number:-
```
<disk type='file' device='disk'>
      <driver name='qemu' type='qcow2'/>
      <source file='/var/lib/libvirt/images/original-test-volume.qcow2'/>
      <target dev='vda' bus='virtio'/>
      <serial>11111111111111111111</serial>
      <address type='pci' domain='0x0000' bus='0x00' slot='0x09' function='0x0'/>
    </disk>
```
But:-
```
udevadm info /sys/class/block/vda
```
retrieves the serial number successfully:-
```
P: /devices/pci0000:00/0000:00:09.0/virtio3/block/vda
N: vda
S: disk/by-id/virtio-11111111111111111111
E: DEVLINKS=/dev/disk/by-id/virtio-11111111111111111111
E: DEVNAME=/dev/vda
E: DEVPATH=/devices/pci0000:00/0000:00:09.0/virtio3/block/vda
E: DEVTYPE=disk
E: ID_SERIAL=11111111111111111111
E: MAJOR=253
E: MINOR=0
E: SUBSYSTEM=block
E: TAGS=:systemd:
E: USEC_INITIALIZED=16682
```
I am working on a patch for this issue but it may take me a while.

**Comment by phillxnet** [comment]
Pull request submitted and ready for review; hopefully relatively uninvasive.

**Comment by phillxnet** [comment]
Closing as patch now merged and released in version 3.8-1
